### PR TITLE
Fix istio multus integration

### PR DESCRIPTION
**Pull request by maiqueb**

### CODE REVIEW

**Comment by maiqueb** [comment]
/cc @ormergi 
/hold let's see how this fares ... 

We can polish the number of tests executed if they pass; right now, only interested in checking the proposed workaround.

File: cluster-provision/k8s/check-cluster-up.sh [hunk 1]
```diff
@@ -100,5 +100,7 @@ export KUBEVIRTCI_GOCLI_CONTAINER=quay.io/kubevirtci/gocli:latest
 
         export SONOBUOY_EXTRA_ARGS="--plugin systemd-logs --plugin e2e"
         hack/conformance.sh $conformance_config
+
+        KUBEVIRT_E2E_FOCUS=istio hack/functests.sh
```
Comment (maiqueb) [inline cluster-provision/k8s/check-cluster-up.sh:104]: @dhiller do we have access to the kubevirt repo from this one ? Is there any easy way to run those tests ?

Comment (maiqueb) [reply cluster-provision/k8s/check-cluster-up.sh]: Perfect, a lot better than my plan A (... which was to clone the kubevirt repo / grab the test binary from the web, and execute it w/ the `focus` on istio. 

Makes sense to me @ormergi . Can you push to this PR ?

Comment (ormergi) [reply cluster-provision/k8s/check-cluster-up.sh]: I posted a draft to test this PR changes with kubevirt istio tests
https://github.com/kubevirt/kubevirtci/pull/919

**Comment by maiqueb** [comment]
> It seems that Istio tests keep failing for a similar reason. job: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/919/check-provision-k8s-1.25/1600093275279069184 On the events logs we see the following error all over:
> 
> ```
> Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_virt-launcher-testvmi-n4d8d-67blr_kubevirt-test-default1_b56da3d9-e063-4ce3-9598-847186bbe52c_0(132f4b7f35b70fc2dd52a6b9caaf613999f092534b694fb245139ce8b455a706): 
> error adding pod kubevirt-test-default1_virt-launcher-testvmi-n4d8d-67blr to CNI network \"k8s-pod-network\": plugin type=\"multus-shim\" name=\"multus-cni-network\" failed (add): 
> CNI request failed with status 400:
> {
>   ContainerID:132f4b7f35b70fc2dd52a6b9caaf613999f092534b694fb245139ce8b455a706
>   Netns:/var/run/netns/1906a5d9-d117-4638-bbd6-5b2d1f800822 IfName:eth0 Args:
>   IgnoreUnknown=1;
>   K8S_POD_NAMESPACE=kubevirt-test-default1;
>   K8S_POD_NAME=virt-launcher-testvmi-n4d8d-67blr;
>   K8S_POD_INFRA_CONTAINER_ID=132f4b7f35b70fc2dd52a6b9caaf613999f092534b694fb245139ce8b455a706;
>   K8S_POD_UID=b56da3d9-e063-4ce3-9598-847186bbe52c 
>   Path: 
>   StdinData:[123 34 99 97 112 97 98 105 108 105 116 105 101 115 34 58 123 34 98 97 110 100 119 105 100 116 104 34 58 116 114 117 101 44 34 112 111 114 116 77 97 112 112 105 110 103 115 34 58 116 114 117 101 125 44 34 99 108 117 115 116 101 114 78 101 116 119 111 114 107 34 58 34 47 104 111 115 116 47 101 116 99 47 99 110 105 47 110 101 116 46 100 47 49 48 45 99 97 108 105 99 111 46 99 111 110 102 108 105 115 116 34 44 34 99 110 105 86 101 114 115 105 111 110 34 58 34 48 46 51 46 49 34 44 34 108 111 103 76 101 118 101 108 34 58 34 118 101 114 98 111 115 101 34 44 34 108 111 103 84 111 83 116 100 101 114 114 34 58 116 114 117 101 44 34 110 97 109 101 34 58 34 107 56 115 45 112 111 100 45 110 101 116 119 111 114 107 34 44 34 116 121 112 101 34 58 34 109 117 108 116 117 115 45 115 104 105 109 34 44 10 32 32 32 32 34 99 104 114 111 111 116 68 105 114 34 58 32 34 47 104 111 115 116 114 111 111 116 34 44 10 32 32 32 32 34 99 111 110 102 68 105 114 34 58 32 34 47 104 111 115 116 47 101 116 99 47 99 110 105 47 110 101 116 46 100 34 44 10 32 32 32 32 34 108 111 103 84 111 83 116 100 101 114 114 34 58 32 116 114 117 101 44 10 32 32 32 32 34 108 111 103 76 101 118 101 108 34 58 32 34 100 101 98 117 103 34 44 10 32 32 32 32 34 108 111 103 70 105 108 101 34 58 32 34 47 116 109 112 47 109 117 108 116 117 115 46 108 111 103 34 44 10 32 32 32 32 34 98 105 110 68 105 114 34 58 32 34 47 111 112 116 47 99 110 105 47 98 105 110 34 44 10 32 32 32 32 34 99 110 105 68 105 114 34 58 32 34 47 118 97 114 47 108 105 98 47 99 110 105 47 109 117 108 116 117 115 34 44 10 32 32 32 32 34 115 111 99 107 101 116 68 105 114 34 58 32 34 47 104 111 115 116 47 114 117 110 47 109 117 108 116 117 115 47 34 10 125 10]
> }
> ERRORED: error configuring pod [kubevirt-test-default1/virt-launcher-testvmi-n4d8d-67blr] networking: 
> Multus: [kubevirt-test-default1/virt-launcher-testvmi-n4d8d-67blr/b56da3d9-e063-4ce3-9598-847186bbe52c]: 
> error loading k8s delegates k8s args: 
> TryLoadPodDelegates: error in getting k8s network for pod: 
> GetNetworkDelegates: failed getting the delegate: 
> GetCNIConfig: err in GetCNIConfigFromFile: no network available in the name istio-cni in cni dir /host/etc/cni/net.d\n'",
> ```

It is strange to see this error: ` no network available in the name istio-cni in cni dir /host/etc/cni/net.d` ... I would expect that when the configuration is using the **chained** option, it would just mutate the default pod network configuration .

**Comment by ormergi** [comment]
> It is strange to see this error: no network available in the name istio-cni in cni dir /host/etc/cni/net.d ... I would expect that when the configuration is using the chained option, it would just mutate the default pod network configuration .

It does, but it seems that Multus expect it to have its own cni conf instead of another plugin as part of k8s-pod-network.

**Comment by phoracek** [comment]
Looking at the gathered logs, something is setting the multus networks annotation requesting istio:

```
.kubevirt.io/name:testvmi-n4d8d] map[cni.projectcalico.org/podIP: cni.projectcalico.org/podIPs: k8s.v1.cni.cncf.io/networks:istio-cni
```

IIUIC, with the chained configuration, this should not happen. Instead, istio would rely on the default pod network config delegating to its CNI.

Is it possible that it's KubeVirt setting this annotation?

**Comment by ormergi** [comment]
> Is it possible that it's KubeVirt setting this annotation?

Yes, in Kubevirt when the VMI spec includes Multus networks it will add the `k8s.v1.cni.cncf.io/networks` annotation to virt-launcher pod with the specified networks.

**Comment by phoracek** [comment]
Ah, ok. So we can say that KubeVirt will not function with chained Istio.

**Comment by maiqueb** [comment]
> Ah, ok. So we can say that KubeVirt will not function with chained Istio?

I can't say for sure; I'm inclined to say yes, you're right - I don't see how it can.

... which basically nullifies the reason for having that exact configuration in https://github.com/kubevirt/kubevirtci/blob/main/cluster-provision/k8s/1.25/istio.sh#L50 (for every provider, when CNAO is not used).

**Comment by maiqueb** [comment]
let's stop this nonsense.

**Comment by maiqueb** [comment]
let's stop this nonsense.

**Pull request closed without merging**